doc/user/spec_examples.md

```diff
@@ -0,0 +1,124 @@
+# Cluster Spec Examples
+
+### Three members cluster
+
+```yaml
+spec:
+  size: 3
+```
+
+This will use default version that etcd operator chooses.
+
+### Three members cluster with version specified
+
+```yaml
+spec:
+  size: 3
+  version: 3.1.1
+```
+
+### Three members cluster with node selector and anti-affinity
+
+```yaml
+spec:
+  size: 3
+  pod:
+    nodeSelector:
+      diskType: ssd
+    antiAffinity: true
+```
+
+### Three members cluster with resource requirement
+
+```yaml
+spec:
+  size: 3
+  pod:
+    resources:
+      limits:
+        cpu: 300m
+        memory: 200Mi
+      requests:
+        cpu: 200m
+        memory: 100Mi
+```
+
+### Three members cluster with PV backup
+
+See [example](../../example/example-etcd-cluster-with-backup.yaml) .
+
+### Three members cluster with S3 backup
+
+```yaml
+spec:
+  size: 3
+  version: "3.1.0"
+  backup:
+    backupIntervalInSecond: 300
+    maxBackups: 5
+    storageType: "S3"
+```
+
+### Three members cluster that restores from previous PV backup
+
+If we created a cluster "cluster-a" and it had backup, but then it was deleted,
+we can restore the same cluster from previous backup:
+
+```yaml
+metadata:
+  name: "cluster-a"
+spec:
+  size: 3
+  version: "3.1.0"
+  backup:
+    backupIntervalInSecond: 300
+    maxBackups: 5
+    storageType: "PersistentVolume"
+    pv:
+      volumeSizeInMB: 512
+  restore:
+    backupClusterName: "cluster-a"
+    storageType: "PersistentVolume"
+```
+
+### Three members cluster that restores from previous S3 backup
+
+Same as above but using "S3" as backup storage.
+
+```yaml
+metadata:
+  name: "cluster-a"
+spec:
+  size: 3
+  version: "3.1.0"
+  backup:
+    backupIntervalInSecond: 300
+    maxBackups: 5
+    storageType: "S3"
+  restore:
+    backupClusterName: "cluster-a"
+    storageType: "S3"
+```
+
+
+### Three members cluster that restores from different cluster's PV backup
+
+If we created a cluster "cluster-a" and it had backup,
```

Comment:
If users want to clone a cluster `cluster-b` from a running cluster `cluster-a` for debugging, testing purpose, use the following example spec: